pennylane/qchem/factorization.py

```diff
@@ -14,39 +14,107 @@
 """
 This module contains the functions needed for two-electron tensor factorization.
 """
-
 from functools import partial
 
 import numpy as np
 import scipy as sp
 
 import pennylane as qml
 
-# pylint: disable=too-many-arguments
-
-
-def factorize(two_electron, tol_factor=1.0e-5, tol_eigval=1.0e-5, cholesky=False):
+has_jax_optax = True
+try:  # pragma: no cover
+    # pylint: disable=unused-import
+    import optax
+    from jax import jit
+    from jax import numpy as jnp
+    from jax import scipy as jsp
+    from jax import value_and_grad
+except (ModuleNotFoundError, ImportError) as e:  # pragma: no cover
+    has_jax_optax = False
+
+# pylint: disable=too-many-arguments, too-many-positional-arguments
+
+
+def factorize(
+    two_electron,
+    tol_factor=1.0e-5,
+    tol_eigval=1.0e-5,
+    cholesky=False,
+    compressed=False,
+    regularization=None,
```

Comment:
We added a lot of arguments and keyword arguments to this function. From 3 arguments to potentially 11 if we use all the `compression_kwargs`

Not necessary for this PR. But is it worth considering another way to structure the `cdf` functionality in the future?

Reply:
We did consider having a separate function for CDF but then decided otherwise to have all the DF functionality contained.